app/src/main/java/org/schabi/newpipe/util/DebounceSaver.java

```diff
@@ -0,0 +1,68 @@
+package org.schabi.newpipe.util;
+
+import org.schabi.newpipe.error.ErrorInfo;
+import org.schabi.newpipe.error.UserAction;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers;
+import io.reactivex.rxjava3.disposables.Disposable;
+import io.reactivex.rxjava3.subjects.PublishSubject;
+
+public class DebounceSaver {
+
+    private final long saveDebounceMillis;
+
+    private final PublishSubject<Long> debouncedSaveSignal;
+
+    private final DebounceSavable debounceSavable;
+
+    // Has the object been modified
+    private final AtomicBoolean isModified;
```

Comment:
I don't get why `isModified` is stored here, as it actually has nothing to do with the debounce saver. Either you move more of the code having to do with it here, or you move it out of this class.

Reply:
This is because `saveChanges` is frequently called by `BookmarkFragment` and `LocalPlaylistFragment`. If `isModified` is not saved in the debounce saver, I need to write 
```java
if (isModified != null) isModified.set(true);
debounceSaver.saveChanges();
```
everytime I want to save changes.

The original method is here. The debounce saver is integrated in `LocalPlaylistFragment` before this pr.
https://github.com/TeamNewPipe/NewPipe/blob/a4037a826822a7cbf56619dd9a782e3583d17287/app/src/main/java/org/schabi/newpipe/local/playlist/LocalPlaylistFragment.java#L617-L624

Reply:
Taking a second look, I think you should change all calls to `setIsModified(false)` to `setNoChangesToSave()` or something like that, to make it clearer that calling such method indicates that we either finished saving, or we just loaded data so nothing needs saving. Feel free to suggest a better name ;-)